File: rp_app/rp_app/customization/salesinvoice.py
import frappe
from frappe import _

def validate_gst_items(doc, method):
    has_exempted = any(i.custom_is_gst_exempt for i in doc.items)
    has_non_exempted = any(not i.custom_is_gst_exempt for i in doc.items)

    if has_exempted and has_non_exempted:
        frappe.throw(_("GST Exempted and Non-Exempted items cannot be billed together."))

File: rp_app/public/js/salesinvoice.js
frappe.ui.form.on("Sales Invoice Item", {
    item_code: function(frm, cdt, cdn) {
        const row = locals[cdt][cdn];
        if (row.item_code) {
            frappe.db.get_value("Item", row.item_code, "custom_is_gst_exempted", (r) => {
                if (r && r.custom_is_gst_exempted !== undefined) {
                    frappe.model.set_value(cdt, cdn, "custom_is_gst_exempt", r.custom_is_gst_exempted);
                }
            });
        }
    }
});

frappe.ui.form.on("Sales Invoice", {
    validate: function(frm) {
        let has_gst_exempted = false;
        let has_non_exempted = false;

        (frm.doc.items || []).forEach(item => {
            const is_exempt = Boolean(item.custom_is_gst_exempt);

            if (is_exempt) {
                has_gst_exempted = true;
            } else {
                has_non_exempted = true;
            }
        });

        if (has_gst_exempted && has_non_exempted) {
            frappe.throw("GST Exempted and Non-Exempted items cannot be billed together. Please create separate invoices.");
        }
    }
});
